src/Components/Aspire.Azure.Messaging.EventHubs/EventProcessorClientComponent.cs

```diff
@@ -23,21 +26,44 @@ protected override void BindClientOptionsToConfiguration(IAzureClientBuilder<Eve
     {
-        return azureFactoryBuilder.RegisterClientFactory<EventProcessorClient, EventProcessorClientOptions>((options, cred) =>
-        {
-            var connectionString = settings.ConnectionString;
-            if (string.IsNullOrEmpty(connectionString) && string.IsNullOrEmpty(settings.Namespace))
+        return azureFactoryBuilder.RegisterClientFactory<EventProcessorClient, EventProcessorClientOptions>(
+            (options, cred) =>
             {
-                throw new InvalidOperationException($"A EventHubProducerClient could not be configured. Ensure valid connection information was provided in 'ConnectionStrings:{connectionName}' or specify a 'ConnectionString' or 'Namespace' in the '{configurationSectionName}' configuration section.");
-            }
+                var connectionString = settings.ConnectionString;
+                if (string.IsNullOrEmpty(connectionString) && string.IsNullOrEmpty(settings.Namespace))
+                {
+                    throw new InvalidOperationException(
+                        $"A EventHubProducerClient could not be configured. Ensure valid connection information was provided in 'ConnectionStrings:{connectionName}' or specify a 'ConnectionString' or 'Namespace' in the '{configurationSectionName}' configuration section.");
+                }
 
-            BlobContainerClient? blobClient = null;
+                if (string.IsNullOrEmpty(settings.BlobClientConnectionName))
+                {
+                    // throw an invalid operation exception if the blob client connection string is not provided
+                    throw new InvalidOperationException(
+                                               $"A EventProcessorClient could not be configured. Ensure valid connection information was provided in 'ConnectionStrings:{connectionName}' or specify a 'BlobClientConnectionName' in the '{configurationSectionName}' configuration section.");
+                }
 
-            return !string.IsNullOrEmpty(connectionString) ?
-                new EventProcessorClient(blobClient, settings.ConsumerGroup, connectionString) :
-                new EventProcessorClient(blobClient, settings.ConsumerGroup, settings.Namespace,
-                    settings.EventHubName, cred, options);
-        }, requiresCredential: false);
+                // todo: add more settings and clientoptions validation; also we should use a deterministic ID for the processor
+
+                var blobConnectionString =
+                    builderConfiguration.GetConnectionString(settings.BlobClientConnectionName);
+
+                Uri blobUri = new Uri($"{blobConnectionString}/checkpoints"); // fixme: this should be a compound name including the consumer group name ideally
+                var blobClient = new BlobContainerClient(blobUri, cred);
+                blobClient.CreateIfNotExists(); // this works, but the processor client fails to create blobs later?
```

Comment:
Instead, this logic should inject the keyed blob client

Reply:
Yeah, I agree - but the plumbing is not there. I added comments above in the main body of the PR (aspire api shortcomings)

Reply:
> Instead, this logic should inject the keyed blob client or non keyed (🤔)

Good luck trying to untangle the IAzureClient* APIs to inject anything supplementary -- it's a jungle of extension methods; otherwise I would have suggested something. It seems to me that some deep work may be needed to allow inter-component dependencies.